MetaMask/metamask-mobile
# [EIP1559] Base for Edit Gas Fee screen

**Description**

This is the base for Edit Gas Fee screen for all the eip1559 UI work. All PRs should branch out of this one and merge to this one.

# QA

Figma: https://www.figma.com/file/NXvUKcZ0Ocu1GPcT7VxRMx/1559---Gas-Controls
Copy: https://docs.google.com/document/d/1qcANOD_E4uvRWcimSCAHmQHJyuPSrHBaiyjryU-7cJk/edit

**Description**

EIP1559 functionalities to QA include:
- transactions
- speed up
- cancel 
- swaps legacy
- transaction history view


Networks to test:
- Mainnet -> should get gas options from legacy API
- EIP-1559 networks (e.g. Ropsten) -> should get gas options from new API
- non EIP-1559 networks (e.g.Kovan) -> should only get a gas price
- Custom networks

Things to test:
- All the different transactions screens: Send flow, dapp transaction, token allowance, any other?
- All the transaction origins: app, dapp, QR, deeplink.
- All the transaction types, send ETH, send ERC-20, send nft
- Smaller screens if everything is reachable by scrolling.

Things to check:
- Check all values (especially gas) on blockchain.
- Check gas values suggested by the dapp or deeplink or QR correspond to the ones on the UI.  You can use https://andrepimenta.github.io/eth_sendTransaction for some of the dapp transactions, good to check if the gas values correspond to the ones suggested by the dapp
- Warnings and validations -> gas too low, too high, insufficient balance, etc.
- When you have a gas selection (low, medium, high), the gas should update every 10 seconds if it has new values. If you have changed the gas on the advanced gas settings, then the gas values should not get updated.
